### ISSUE TITLE
Event when everything loaded including tokens if applicable?

### ISSUE THREAD

**Issue by JohnGalt1717** [damienbod/angular-auth-oidc-client#257]
Hi there.

I'm attempting to set up our system so that if the user is already logged in they redirect to their home screen. But only if they are already logged in on the first visit.

This should be trivial but is deceptively difficult.

The onModuleSetup method calls before everything is loaded so you can't subscribe to that.

If you subscribe to the rest of the events there is no consistency to when they fire.

i.e. they will fire every time the site is reloaded even on login completions. (and getIsAuthorized() as an example fires twice on startup, once with false and once with true if logged in, so there is no way to know if the site is done loading completely because if they're not logged in you only get the false)

I'm trying to trap a way that ignores login completions because obviously that should be handled separately and just handle the user coming back to the site with a fresh session and the login info is there.

Suggestions?

**Comment by profet23** [comment]
I use this in order to check authorization:

```
this.isAuthorized$ = new Observable<boolean>((subscriber) => {
            let subscription = this.oidcSecurityService.onModuleSetup.pipe(take(1)).subscribe(() => {
                    subscriber.next(true);
                    subscriber.complete();
                });

            if (this.oidcSecurityService.moduleSetup) {
                subscriber.next(true);
                subscriber.complete();
                subscription.unsubscribe();
            }
        }).pipe(
            switchMap(() => {
                let race$ = race<any>(
                    this.oidcSecurityService.getIsAuthorized().pipe(filter((isAuthorized) => isAuthorized == true)), 
                    this.oidcSecurityService.onAuthorizationResult.asObservable()     
                );

                this.oidcSecurityService.refreshSession();
                
                return race$;
            }),
            switchMapTo(this.oidcSecurityService.getIsAuthorized()),
            shareReplay(1)
        );
```

@damienbod  I feel like this comes up a lot. maybe something like the above needs to be incorporated?

**Comment by JohnGalt1717** [comment]
@profet23 @damienbod 100% . This needs to be a standard event that is fired after all of this. And there needs to be an option to refresh an expired token before this is fired so that it will go and try and do a silent redirect through (if the login site uses cookies and skips through) and then and only then fires this, and can automatically redirect to login if unable to refresh it that way.

What you want is a site to ONLY show up in the logged in or not state. This event should fire passing false if not logged in after all of this stuff too. We would then use this to kick off the UI display and loads of data from there according to logged in or not. 

Basically at the end of this event that occurs on bootstrap you should conclusively know if the user is logged in or not AND if the user was logged in before but refresh failed. (i.e. 3 possible result states)  The final result state just mentioned would, in our case and most others, result in an automatic redirect to login again.

**Comment by damienbod** [comment]
fixed in version 6.0.12
